app/controllers/concerns/two_factor_authenticatable.rb

```diff
@@ -78,9 +81,19 @@ def render_show_after_invalid
 
   def update_invalid_user
     current_user.second_factor_attempts_count += 1
-    # set time lock if max attempts reached
-    current_user.second_factor_locked_at = Time.zone.now if current_user.max_login_attempts?
-    current_user.save
+
+    UpdateUser.new(
+      user: current_user,
+      attributes: set_time_lock_if_max_attempts_reached
```

Comment:
I think you missed the `second_factor_attempts_count` attribute. Here, we definitely want to increment `second_factor_attempts_count` by 1, and we want to use `UpdateUser` to do that. Then, optionally, we might also want to update `second_factor_locked_at`. So, I think perhaps this will do it:
```ruby
attributes = { second_factor_attempts_count: current_user.second_factor_attemps_count + 1 }
attributes[:second_factor_locked_at] = Time.zone.now if current_user.max_login_attempts?

UpdateUser.new(
  user: current_user, 
  attributes: attributes
)
```